tests/components/rfxtrx/conftest.py

```diff
@@ -0,0 +1,75 @@
+"""Common test tools."""
+import threading
+from unittest import mock
+
+import RFXtrx
+import pytest
+
+from homeassistant.components import rfxtrx as rfxtrx_core
+
+from tests.common import mock_component
+
+
+class FixedDummySerial(RFXtrx._dummySerial):  # pylint: disable=protected-access
+    """Fixed dummy serial that doesn't cause max CPU usage."""
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+        self._close_event = threading.Event()
+
+    def read(self, data=None):
+        """Read."""
+        res = super().read(data)
+        if not res and not data:
+            self._close_event.wait(0.1)
+        return res
+
+    def close(self):
+        """Close."""
+        self._close_event.set()
+
+
+class FixedDummyTransport(RFXtrx.DummyTransport):
+    """Fixed dummy transport that maxes CPU."""
+
+    def __init__(self, device="", debug=True):
+        """Init."""
+        super().__init__(device, debug)
+        self._close_event = threading.Event()
+
+    def receive_blocking(self, data=None):
+        """Read."""
+        res = super().receive_blocking(data)
+        if not res:
+            self._close_event.wait(0.1)
+        return res
+
+    def close(self):
+        """Close."""
+        self._close_event.set()
```

Comment:
This is quite ugly. I'll see if i can re-work this when i get to refactoring tests. This is the background to the slowdowns the test caused. It used totally non blocking receive functions in a while True loop. So the threads just eats up CPU. As well has a run-away thread, not cleaned up.

Reply:
Do we need a real transport? We can't just patch it?

Reply:
We are patching a dummy transport, designed for testing.

Reply:
Sure, but if that's causing problem, is it necessary?

Reply:
We could replace it fully. But that is sort of what we already do. I don't really want to add device specific bytestreams to our code (also I don't understand the protocol enough yet)

Reply:
What do we need the transport to do in the tests? Does it need to return bytesstreams to complete the tests? Normally we only need to test the interface between the library and home assistant.